tests/kernel/timer/timer_monotonic/src/main.c

```diff
@@ -8,21 +8,132 @@
 #include <tc_util.h>
 #include <ztest.h>
 
-int test_frequency(void)
+static struct k_timer test_timer;
+static struct k_timer test_periodtimer;
+
+u32_t start_time, stop_time, cycles_spent, nanoseconds_spent,
+	   time_spent[10], cycles_cost[10], pctlist[10];
+int count = 0;
+
+
+
+static void callback_period(struct k_timer *timer)
+{
+	u32_t pct;
+
+	stop_time = k_cycle_get_32();
+	cycles_spent = stop_time - start_time;
+	nanoseconds_spent = (u32_t)k_cyc_to_ns_floor64(cycles_spent);
+	pct = (u64_t)cycles_spent * 100000U / sys_clock_hw_cycles_per_sec();
```

Comment:
Naming: "pct" probably meant "percent" in the original code, but this is producing a value in units of 1/100000.